core/impl/src/main/java/pt/webdetails/cdf/dd/util/Utils.java

```diff
@@ -170,14 +171,18 @@ public static Document getDocument( InputStream input ) throws DocumentException
     return reader.read( input );
   }
 
-  @SuppressWarnings( "unchecked" )
-  public static List<Element> selectNodes( Element elem, String xpath ) {
-    return elem.selectNodes( xpath );
+  public static List<Element> selectNodes( Element element, String xpath ) {
+    return ( (List<Node>) element.selectNodes( xpath ) ).stream()
```

Comment:
Again, I insist. Instead of copying into a new list to, downcast each element, you can simply cast it this way (as described in https://stackoverflow.com/questions/15427624/could-not-downcast-using-list-class-in-java/15427707#15427707):
```java
return (List<Element>) ( (List<?>) element.selectNodes( xpath ) ) );
```

Also, it would probably be a good idea to rename the method to `selectElements`. The correctness of the cast depends on the nature of specified `xpath`, so people better be aware that the `xpath` must select element nodes.